tg_bot/modules/welcome.py

```diff
@@ -129,118 +130,127 @@ def cas_banned(userid):
+                    )
 
-            else:
-                # BEGINNING THE BAN
-                log = (
-                    "<b>CAS BAN:</b>" "\n#CASBAN" "\n<b>User:</b> {}".format(new_mem.id)
-                )
+                    reason = "Ban via CAS api query"
 
-                reason = "Ban via CAS api query"
+                    if reason:
+                        log += "\n<b>Motivo:</b> {}".format(reason)
 
-                if reason:
-                    log += "\n<b>Motivo:</b> {}".format(reason)
+                    user_id = new_mem.id
+                    try:
+                        chat.kick_member(user_id)
+                        bot.send_sticker(chat.id, BAN_STICKER)  # banhammer electus sticker
+                        message.reply_text("BANNATO VIA SISTEMA AUTOMATICO CAS!")
+                        return log
 
+                    except BadRequest as excp:
+                        if excp.message == "Reply message not found":
+                            # Do not reply
+                            message.reply_text("BANNATO!", quote=False)
+                            return log
+                        else:
+                            LOGGER.warning(update)
+                            LOGGER.exception(
+                                "ERROR in CAS banning user %s in chat %s (%s) due to %s",
+                                user_id,
+                                chat.title,
+                                chat.id,
+                                excp.message,
+                            )
+                            message.reply_text(
+                                "Diamine, non riesco a bannare questo utente."
+                            )
+            else:
+                # Kicking the user because of the username
                 user_id = new_mem.id
-                try:
-                    chat.kick_member(user_id)
-                    bot.send_sticker(chat.id, BAN_STICKER)  # banhammer electus sticker
-                    message.reply_text("BANNATO VIA SISTEMA AUTOMATICO CAS!")
-                    return log
-
-                except BadRequest as excp:
-                    if excp.message == "Reply message not found":
-                        # Do not reply
-                        message.reply_text("BANNATO!", quote=False)
-                        return log
-                    else:
-                        LOGGER.warning(update)
-                        LOGGER.exception(
-                            "ERROR in CAS banning user %s in chat %s (%s) due to %s",
-                            user_id,
-                            chat.title,
-                            chat.id,
-                            excp.message,
-                        )
-                        message.reply_text(
-                            "Diamine, non riesco a bannare questo utente."
-                        )
+                chat.kick_member(user_id, until_date=1602201900)
```

Comment:
`until_date` dovrebbe essere dinamico

Reply:
`until_date` prende in input tempo unix, dovrebbe essere calcolato nel seguente modo se vogliamo sbloccare dopo cinque minuti:

`until_date=time.time() + 300`

Reply:
forse non ho pushato, devo ricontrollare

Reply:
Aggiornato!